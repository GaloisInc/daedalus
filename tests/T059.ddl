import T059_A

def P1 = F UInt8
def P2 = F P1

def Main = { P1; P2 }
