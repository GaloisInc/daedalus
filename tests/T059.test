dump
--spec
T059.ddl
