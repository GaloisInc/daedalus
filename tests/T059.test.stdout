module DaedalusMain
 
--- Imports:
 
--- Type defs:
 
--- Rules:
 
T059.F_14 : Grammar (uint 8) =
  Match UInt8
 
T059.P1 : Grammar (uint 8) =
  T059.F_14
 
T059.F_13 : Grammar (uint 8) =
  T059.P1
 
T059.P2 : Grammar (uint 8) =
  T059.F_13
 
T059.F__12 : Grammar {} =
  @Match UInt8
 
T059.P1_ : Grammar {} =
  T059.F__12
 
T059.Main : Grammar (uint 8) =
  do T059.P1_
     ($$ : uint 8) <- T059.P2
     pure $$
